src/components/Inputs/GenericInput.tsx

```diff
@@ -0,0 +1,143 @@
+import styled from '@emotion/styled';
+import React from 'react';
+import { pxToRem } from '../../utils';
+
+export type GenericLabelPositions = 'top' | 'right' | 'bottom' | 'left';
+export type GenericTypes = 'text' | 'email' | 'password';
+
+export interface GenericInputProps {
+  readonly inputType: GenericTypes;
+  readonly label?: string;
+  readonly labelPosition?: GenericLabelPositions;
+  readonly gapSize?: number;
+  readonly fontSize?: number;
+  readonly fontColor?: string;
+  readonly placeholder?: string;
+  readonly placeholderColor?: string;
+  readonly boxShadowColor?: string;
+  readonly borderColor?: string;
+  readonly focusBorderColor?: string;
+  readonly focusBorderWidth?: number;
+  readonly width?: number;
+  readonly height?: number;
+  readonly minTextLength?: number;
+  readonly maxTextLength?: number;
+  readonly spellcheck?: boolean;
+  readonly pattern?: string;
+  readonly disabled?: boolean;
+  readonly required?: boolean;
+  readonly onChange?: React.ChangeEventHandler<HTMLInputElement>;
+}
+
+export const GenericInput: React.FC<GenericInputProps> = ({
+  inputType = 'tel',
```

Comment:
does 'tel' work even though it is not part of the GenericTypes ?

Reply:
Good catch!  That's a typo.  I'm not including that type.